core/object/script_language_extension.h

```diff
@@ -650,9 +650,22 @@ class ScriptInstanceExtension : public ScriptInstance {
 			const GDExtensionPropertyInfo *pinfo = native_info->get_property_list_func(instance, &pcount);
 
 #ifdef TOOLS_ENABLED
-			Ref<Script> script = get_script();
-			if (script.is_valid() && pcount > 0) {
-				p_list->push_back(script->get_class_category());
+			if (native_info->get_class_category_func) {
+				uint32_t pcount_cc;
+				const GDExtensionPropertyInfo *cc = native_info->get_class_category_func(instance, &pcount_cc);
```

Comment:
Does it really make sense for this to return multiple class categories? The other code branch only adds one.

And if it only needs to do one, I think we could do this without needing to have the `free_class_category_func`. The function signature could be changed to:

```
typedef GDExtensionBool (*GDExtensionScriptInstanceGetClassCategory)(GDExtensionScriptInstanceDataPtr p_instance, GDExtensionPropertyInfo *p_class_category);
```

... and then this code could look like:

```
GDExtensionPropertyInfo gdext_class_category;
if (native_info->get_class_category_func(instance, &gdext_class_category)) {
  p_list->push_back(PropertyInfo(gdext_class_category));
}
```

Reply:
I must admit understanding why this needed to be a list was and is beyond my skill level.

Reply:
Hm, well, I get the feeling that it doesn't need to be a list. :-) If you don't have a reason for it, I'd recommend switching to use the code in my previous comment which gets only a single category via this new API.

Reply:
Thanks for the suggestion.